File: src/main/java/org/adoptopenjdk/jheappo/objects/UTF8String.java
package org.adoptopenjdk.jheappo.objects;

import org.adoptopenjdk.jheappo.heapdump.EncodedChunk;

public class UTF8String extends HeapObject {

    private String string;

    public UTF8String(EncodedChunk buffer) {
        super(buffer);
        string = new String(buffer.readRemaining());
    }

    public String getString() {
        return string;
    }

    public String toString() { return getId() + " : " + string; }
}

File: src/main/java/org/adoptopenjdk/jheappo/objects/UTF8String.kt
package org.adoptopenjdk.jheappo.objects

import org.adoptopenjdk.jheappo.heapdump.EncodedChunk

class UTF8String(buffer: EncodedChunk) : HeapObject(buffer) {

    val string: String = String(buffer.readRemaining())

    override fun toString(): String {
        return "$id : $string"
    }
}
